### ISSUE TITLE
.js wasm loader does not work from non-cargo web server without application/wasm support

### ISSUE THREAD

**Issue by pythagorean** [koute/cargo-web#192]
If I use `cargo web deploy` and an external web server that does not have an application/wasm MIME type defined or able to be configured, then a browser trying to load the wasm may fail.

Here is the error displayed by Chrome in its console:

`Uncaught (in promise) TypeError: Failed to execute 'compile' on 'WebAssembly': Incorrect response MIME type. Expected 'application/wasm'.`

I have been able to make a workaround by patching the JS file as follows:

```
--- my_app.js.old
+++ my_app.js
@@ -27,11 +27,7 @@
         } else {
             var file = fetch( "my_app.wasm", {credentials: "same-origin"} );
 
-            var wasm_instance = ( typeof WebAssembly.instantiateStreaming === "function"
-                ? WebAssembly.instantiateStreaming( file, instance.imports )
-                    .then( function( result ) { return result.instance; } )
-
-                : file
+            var wasm_instance = ( file
                     .then( function( response ) { return response.arrayBuffer(); } )
                     .then( function( bytes ) { return WebAssembly.compile( bytes ); } )
                     .then( function( mod ) { return WebAssembly.instantiate( mod, instance.imports ) } ) );
```

**Comment by koute** [comment]
Yes. This is expected, and unless someone has a suggestion which doesn't involve turning off streaming I don't think I can do anything about it - it's the browser who's rejecting the WASM module after all.

**Comment by Pauan** [comment]
(Also, you can simply switch to a web server which *does* support `application/wasm`, such as [https](https://crates.io/crates/https))

**Comment by pythagorean** [comment]
In the instant case I may need to use a deployment web server that is not able to provide this facility. I understand streaming may not be supported in this case but there should be some better test and reversion to the second method rather than having to patch the loader code manually.

**Comment by Pauan** [comment]
Yeah, I guess it could add in a `.catch` which falls back to `WebAssembly.compile`.

That would incur a bit of extra performance loss in the failure case (at least one microtask cycle), but at least it would work.
